src/util/codec/mysql/decimal.rs

```diff
@@ -184,6 +195,22 @@ fn sub(lhs: u32, rhs: u32, carry: &mut i32, res: &mut u32) {
     }
 }
 
+/// `sub2` substracts rhs and carry from lhs, stores the diff and new carry.
+/// the new carry may be 2.
+fn sub2(lhs: u32, rhs: u32, carry: &mut i32, res: &mut u32) {
+    let mut diff = lhs as i32 - rhs as i32 - *carry;
+    if diff < -(WORD_BASE as i32) {
```

Comment:
This is not the same with TiDB decimal. https://github.com/pingcap/tidb/blob/master/mysql/mydecimal.go#L91